src/inventory_client/inventory_client.go

```diff
@@ -80,6 +81,11 @@ func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret str
 			RootCAs:            certs,
 		},
 	})
+	// Add retry settings
+	clientConfig.Transport = rehttp.NewTransport(transport,
+		rehttp.RetryMaxRetries(10),
+		rehttp.ExpJitterDelay(2, time.Duration(10)*time.Minute),
```

Comment:
Formatting :)

Reply:
2048 nanoseconds (=2.048 microseconds=0.002048 ms) seem too short. Not comparable even to a typical round-trip time. Here I simulated a progression of delays with the values in your code.
```
attempt: 1, delay: 0s
attempt: 2, delay: 3ns
attempt: 3, delay: 0s
attempt: 4, delay: 9ns
attempt: 5, delay: 11ns
attempt: 6, delay: 79ns
attempt: 7, delay: 89ns
attempt: 8, delay: 453ns
attempt: 9, delay: 138ns
attempt: 10, delay: 1.672µs
```